### PR TITLE
improve select_build_worker

### PR DESCRIPTION
**Pull request by Matuiss2**
- Make it prefer workers that are not carrying any resource and idle drones, if any are like this(very rare cases) select a random drone(if forced)

### CODE REVIEW

File: sc2/bot_ai.py [hunk 1]
```diff
@@ -304,16 +304,16 @@ def can_afford(self, item_id: Union[UnitTypeId, UpgradeId, AbilityId], check_sup
...
-        """Select a worker to build a bulding with."""
+        """Select a worker to build a building with."""
 
-        workers = self.workers.closer_than(20, pos) or self.workers
-        for worker in workers.prefer_close_to(pos).prefer_idle:
-            if not worker.orders or len(worker.orders) == 1 and worker.orders[0].ability.id in {AbilityId.MOVE,
-                                                                                                AbilityId.HARVEST_GATHER,
-                                                                                                AbilityId.HARVEST_RETURN}:
-                return worker
+        workers = self.workers.gathering.closer_than(20, pos) | self.workers.idle or self.workers
```
Comment (tweakimp) [inline sc2/bot_ai.py]: Please make one units call out of this by using `self.works.filter`

Comment (Matuiss2) [reply sc2/bot_ai.py]: hopefully fixed

**Comment by tweakimp** [comment]
Looks good to me.

**Comment by Dentosal** [comment]
This is going to be merged to develop first. Please resolve the conflicts if possible.

**Pull request closed without merging**